sink_test.go

```diff
@@ -118,3 +120,77 @@ func TestFanoutSink_Sample(t *testing.T) {
 		t.Fatalf("val not equal")
 	}
 }
+
+func TestNewMetricSinkFromURL(t *testing.T) {
```

Comment:
I think we need to restructure this test a bit. While it does cover the URL-enabled sinks broadly, we are skipping some of the depth. I would just have this function ensure that the output is a sink of the expected type, and then test the `NewXXXSinkFromURL`'s in isolation. This would also make it clearer because they all have different possibilities of returned errors.